### PR TITLE
update devops api doc & remove not used api

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

File: pkg/apis/devops/v1alpha2/register.go [hunk 1]
```diff
@@ -390,7 +390,7 @@ func addWebService(c *restful.Container) error {
 		Writes(devops.StopPipe{}))
 
 	// match /blue/rest/organizations/jenkins/pipelines/{projectName}/pipelines/{pipelineName}/runs/{runId}/stop/
-	webservice.Route(webservice.PUT("/devops/{projectName}/pipelines/{pipelineName}/runs/{runId}/stop").
+	webservice.Route(webservice.POST("/devops/{projectName}/pipelines/{pipelineName}/runs/{runId}/stop").
```
Comment (runzexia) [inline pkg/apis/devops/v1alpha2/register.go]: /devops/{devops}/pipelines/{pipeline}/runs/{run}/stop

File: pkg/apis/devops/v1alpha2/register.go [hunk 2]
```diff
@@ -738,30 +708,30 @@ func addWebService(c *restful.Container) error {
...
-		Doc("Get notification commit by HTTP GET method.").
+		Doc("Get commit notification by HTTP GET method.").
 		Produces("text/plain; charset=utf-8").
 		Param(webservice.QueryParameter("url", "url of git scm").
 			Required(true).
 			DataFormat("url=%s")))
 
 	// Gitlab or some other scm managers can only use HTTP method. match /git/notifyCommit/?url=
-	webservice.Route(webservice.POST("/devops/notifycommit").
+	webservice.Route(webservice.POST("/notifycommit").
```
[the remark this answers is not in the document]
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: /git/webhook/notifycommit ?

File: pkg/apiserver/devops/devops.go [hunk 3]
```diff
@@ -244,7 +244,7 @@ func ReplayPipeline(req *restful.Request, resp *restful.Response) {
 }
 
 func GetBranchRunLog(req *restful.Request, resp *restful.Response) {
-	projectName := req.PathParameter("projectName")
+	projectName := req.PathParameter("devops")
 	pipelineName := req.PathParameter("pipelineName")
```
Comment (runzexia) [inline pkg/apiserver/devops/devops.go]: pipelineName & brachName also need to change

File: pkg/apiserver/devops/devops.go [hunk 4]
```diff
@@ -617,7 +584,7 @@ func GetBranchNodesDetail(req *restful.Request, resp *restful.Response) {
 }
 
 func GetNodesDetail(req *restful.Request, resp *restful.Response) {
-	projectName := req.PathParameter("projectName")
+	projectName := req.PathParameter("devops")
 	pipelineName := req.PathParameter("pipelineName")
 	runId := req.PathParameter("runId")
```
Comment (runzexia) [inline pkg/apiserver/devops/devops.go]: runId

File: pkg/apis/devops/v1alpha2/register.go [hunk 5]
```diff
@@ -216,18 +216,18 @@ func addWebService(c *restful.Container) error {
 		Returns(http.StatusOK, RespOK, []devops.JenkinsCredential{}).
 		Reads([]devops.JenkinsCredential{}))
 
-	// match Jenkisn api "/blue/rest/organizations/jenkins/pipelines/{projectName}/{pipelineName}"
+	// match Jenkisn api "/blue/rest/organizations/jenkins/pipelines/{devops}/{pipelineName}"
 	webservice.Route(webservice.GET("/devops/{devops}/pipelines/{pipelines}").
```
Comment (rayzhou2017) [inline pkg/apis/devops/v1alpha2/register.go]: still keep /devops/?

File: pkg/apis/devops/v1alpha2/register.go [hunk 6]
```diff
@@ -216,18 +216,18 @@ func addWebService(c *restful.Container) error {
...
 		To(devopsapi.GetPipeline).
 		Metadata(restfulspec.KeyOpenAPITags, tags).
 		Doc("Get a Pipeline Inside a DevOps Project").
 		Param(webservice.PathParameter("pipelines", "the name of pipeline, which helps to deliver continuous integration continuous deployment.")).
```
Comment (rayzhou2017) [inline pkg/apis/devops/v1alpha2/register.go]: the path should be pipelineName, right?

File: pkg/apis/devops/v1alpha2/register.go (hunk 5 above)
Comment (runzexia) [reply pkg/apis/devops/v1alpha2/register.go]: this devops means devops project

File: pkg/apis/devops/v1alpha2/register.go (hunk 6 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: here should be pipeline, and all api schema is "${}s/${}".

File: pkg/apis/devops/v1alpha2/register.go [hunk 7]
```diff
@@ -265,97 +265,97 @@ func addWebService(c *restful.Container) error {
...
 		Param(webservice.QueryParameter("credentialId", "credential id for SCM.").
 			Required(true).
 			DataFormat("credentialId=%s")).
 		Returns(http.StatusOK, RespOK, []devops.SCMOrg{}).
 		Writes([]devops.SCMOrg{}))
 
-	// match "/blue/rest/organizations/jenkins/scm/{scmId}/organizations/{organizationId}/repositories/?credentialId=&pageNumber&pageSize="
-	webservice.Route(webservice.GET("/devops/scm/{scmId}/organizations/{organizationId}/repositories").
+	// match "/blue/rest/organizations/jenkins/scm/{scm}/organizations/{organizationId}/repositories/?credentialId=&pageNumber&pageSize="
+	webservice.Route(webservice.GET("/scms/{scm}/organizations/{organizationId}/repositories").
```
Comment (runzexia) [inline pkg/apis/devops/v1alpha2/register.go]: organizationId -》 organization

File: pkg/apis/devops/v1alpha2/register.go [hunk 8]
```diff
@@ -514,209 +514,179 @@ func addWebService(c *restful.Container) error {
...
-		Param(webservice.PathParameter("stepId", "pipeline step id")))
+		Param(webservice.PathParameter("devops", "DevOps Project's Id, e.g. project-RRRRAzLBlLEm")).
+		Param(webservice.PathParameter("pipeline", "the name of pipeline, which helps to deliver continuous integration continuous deployment.")).
+		Param(webservice.PathParameter("run", "pipeline run id, the unique id for a pipeline once build.")).
+		Param(webservice.PathParameter("node", "pipeline node id, the one node in pipeline.")).
+		Param(webservice.PathParameter("step", "pipeline step id")))
 
 	// match /job/project-8QnvykoJw4wZ/job/test-1/indexing/consoleText
-	webservice.Route(webservice.GET("/devops/{projectName}/pipelines/{pipelineName}/console/log").
+	webservice.Route(webservice.GET("/devops/{devops}/pipelines/{pipeline}/console/log").
```
Comment (runzexia) [inline pkg/apis/devops/v1alpha2/register.go]: console/log -> consolelog

File: pkg/apis/devops/v1alpha2/register.go (hunk 7 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: fix comment

File: pkg/apis/devops/v1alpha2/register.go (hunk 6 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: fix comment

File: pkg/apis/devops/v1alpha2/register.go (hunk 8 above)
Comment (soulseen) [reply pkg/apis/devops/v1alpha2/register.go]: fix

**Pull request merged**